ext/standard/math.c

```diff
@@ -389,77 +389,38 @@ PHP_FUNCTION(round)
 }
 /* }}} */
 
-/* {{{ Returns the sine of the number in radians */
-PHP_FUNCTION(sin)
-{
-	double num;
-
-	ZEND_PARSE_PARAMETERS_START(1, 1)
-		Z_PARAM_DOUBLE(num)
-	ZEND_PARSE_PARAMETERS_END();
-	RETURN_DOUBLE(sin(num));
-}
-/* }}} */
-
-/* {{{ Returns the cosine of the number in radians */
-PHP_FUNCTION(cos)
-{
-	double num;
-
-	ZEND_PARSE_PARAMETERS_START(1, 1)
-		Z_PARAM_DOUBLE(num)
-	ZEND_PARSE_PARAMETERS_END();
-	RETURN_DOUBLE(cos(num));
-}
-/* }}} */
-
-/* {{{ Returns the tangent of the number in radians */
-PHP_FUNCTION(tan)
-{
-	double num;
+// Lots of the PHP function are just wrappers around a C math function; if they
+// - are named the same as the C implementatino
+// - take a single parameter, a double
+// - return a double
+// then they can use this macro for de-duplication
+#define MATH_PHP_FUNCTION_WRAPPER(name) \
+	PHP_FUNCTION(name) \
+	{ \
+		double num; \
+		ZEND_PARSE_PARAMETERS_START(1, 1) \
+			Z_PARAM_DOUBLE(num) \
+		ZEND_PARSE_PARAMETERS_END(); \
+		RETURN_DOUBLE(name(num)); \
+	}
```

Comment:
I really don't like using macro in this way as it makes debugging uglier. This is really not an improvement and should be taken off.